packages/payment-proxy-client/src/file.ts

```diff
@@ -38,3 +33,91 @@ function parseFileNameFromUrl(url: string): string {
     return "fetched-file";
   }
 }
+
+export async function fetchFileInChunks(
+  chunkSize: number,
+  baseUrl: string,
+  costPerByte: number,
+  selectedChannel: string,
+  nitroClient: NitroRpcClient
+): Promise<File> {
+  const firstChunk = await fetchFileChunk(
+    0,
+    chunkSize - 1,
+    baseUrl,
+    costPerByte,
+    selectedChannel,
+    nitroClient
+  );
+
+  const { contentLength, fileName } = firstChunk;
+  const remainingChunks = Math.ceil(contentLength / chunkSize) - 1;
+
+  const fileContents = new Uint8Array(contentLength);
+  fileContents.set(firstChunk.data);
+
+  for (let i = 1; i <= remainingChunks; i++) {
+    const start = i * chunkSize;
+    const stop = start + chunkSize - 1;
+
+    const { data } = await fetchFileChunk(
+      start,
+      stop,
+      baseUrl,
+      costPerByte,
+      selectedChannel,
+      nitroClient
+    );
+
+    fileContents.set(data, i * chunkSize);
+  }
+
+  return new File([fileContents], fileName);
+}
+
+export async function fetchFileChunk(
+  start: number,
+  stop: number,
+  baseUrl: string,
+  costPerByte: number,
+  selectedChannel: string,
+  nitroClient: NitroRpcClient
+): Promise<{ data: Uint8Array; contentLength: number; fileName: string }> {
+  const dataLength = stop - start + 1; // +1 because stop is inclusive
+  console.log(dataLength);
+  const chunkCost = dataLength * costPerByte;
+
+  const voucher = await nitroClient.CreateVoucher(selectedChannel, chunkCost);
+
+  const req = new Request(
+    `${baseUrl}?channelId=${voucher.ChannelId}&amount=${voucher.Amount}&signature=${voucher.Signature}`
+  );
```

Comment:
1. Consider factoring into a helper function that accepts `voucher` to aid future refactoring. 
2. Should we consider optimising the format of these requests somewhat? i.e. using `c`,`a`,`s` for the variables

Reply:
I see you basically already did 1, so ignore that (commented on an intermediate commit).

Reply:
> Should we consider optimising the format of these requests somewhat? i.e. using c,a,s for the variables

I personally prefer the longer terms as it makes the query string a lot more readable. What would we be optimizing for?

Reply:
Just the size of the request. But we don't stand to trim it down by much and it also probably doesn't matter much either 🤷